### PR TITLE
Feature enable sequence based metadata creation

### PR DESCRIPTION
**Pull request by Etschbeijer**
Closses #447

### CODE REVIEW

File: src/Spreadsheet/ArcAssay.fs [hunk 1]
```diff
@@ -76,13 +76,70 @@ module ArcAssay =
...
+                yield SparseRow.fromValues [contactsLabel]
+                yield! Contacts.toRows (Some contactsPrefix) (List.ofSeq assay.Performers)
+            }
+        assay
+        |> toRows
+        |> Seq.map (fun row -> SparseRow.getAllValues row)
+
+    let fromMetadataCollection (collection: seq<seq<string option>>) : ArcAssay =
+        try
+            let fromRows (usePrefixes: bool) (rows: seq<SparseRow>) =
```
Comment (HLWeil) [inline src/Spreadsheet/ArcAssay.fs:94]: Pull this (fromRows) function out to be used both in `fromMetadataSheet` and your function.

File: src/Spreadsheet/ArcAssay.fs [hunk 2]
```diff
@@ -76,13 +76,70 @@ module ArcAssay =
...
+                if en.MoveNext () then
+                    let currentLine = en.Current |> SparseRow.tryGetValueAt 0
+                    loop currentLine [] [] 1
+            
+                else
+                    failwith "empty assay metadata sheet"
+            let rows =        
+                collection 
+                |> Seq.map SparseRow.fromAllValues
+            let hasPrefix =
```
Comment (HLWeil) [inline src/Spreadsheet/ArcAssay.fs:128]: This hasPrefix part can be moved into the `fromRows` function and dropped as a param there.

**Review by HLWeil — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please reuse the `fromRows` function to reduce code duplication. 

And please undo the styling changes not connected to the specific functions you change, to keep the PR more clean. 
Also I would prefer to keep the `param : type` styling in general. With the `,`, I'm not sure yet. But I heavily advocate against including this kind of cleanups of code-parts unrelated to the PR.

File: src/Spreadsheet/ArcAssay.fs (hunk 2 above)
Comment (Etschbeijer) [reply src/Spreadsheet/ArcAssay.fs]: fixed

File: src/Spreadsheet/ArcAssay.fs (hunk 1 above)
Comment (Etschbeijer) [reply src/Spreadsheet/ArcAssay.fs]: fixed

**Comment by Etschbeijer** [comment]
> Thanks for the PR!
> 
> Please reuse the `fromRows` function to reduce code duplication.
> 
> And please undo the styling changes not connected to the specific functions you change, to keep the PR more clean. Also I would prefer to keep the `param : type` styling in general. With the `,`, I'm not sure yet. But I heavily advocate against including this kind of cleanups of code-parts unrelated to the PR.

Fixed the parameter styling and I will avoid style changes in the futur.

**Review by HLWeil — APPROVED** [review]
Thanks, looks good now!

**Pull request merged**